react/src/views/Transfers/CreateShipment/CreateShipmentView.tsx

```diff
@@ -0,0 +1,250 @@
+        ({
+          id: agreement.id,
+          name: agreement.name,
+          bases: agreement.bases,
+        } as IOrganisationBaseData),
+    )
+    .reduce((accumulator, currentOrg) => {
+      // Merge options. If there are multiple transfer agreements this step is necessary
+      const existingOrganisation = accumulator.find((org) => org.id === currentOrg.id);
+      if (existingOrganisation) {
+        existingOrganisation.bases.push(
+          ...currentOrg.bases.filter(
+            (base) => !existingOrganisation.bases.some((b) => b.id === base.id),
+          ),
+        );
+      } else {
+        accumulator.push(currentOrg);
+      }
+      return accumulator;
+    }, [] as IOrganisationBaseData[]);
+
+  // Handle Submission
+  const onSubmitCreateShipmentForm = useCallback(
+    (createShipmentFormData: ICreateShipmentFormData) => {
+      // Find the possible agreement Ids for the partner base
+      const agreementIds: Array<string> =
+        acceptedTransferAgreementsPartnerData
+          ?.filter((org) =>
+            org.bases.some((base) => base.id === createShipmentFormData.receivingBase.value),
+          )
+          .map((org) => org.agreementId) || [];
+
+      if (agreementIds.length === 0) {
+        triggerError({
+          message: "Error while trying to create a new shipment",
+        });
+      } else {
+        createShipmentMutation({
+          variables: {
+            // This is just a hack since it is possible that multiple agreements exist for the same base
+            transferAgreementId: parseInt(agreementIds[0], 10),
+            sourceBaseId: parseInt(baseId, 10),
+            targetBaseId: parseInt(createShipmentFormData.receivingBase.value, 10),
+          },
+        })
+          .then((mutationResult) => {
+            if (mutationResult.errors) {
+              triggerError({
+                message: "Error while trying to create a new shipment!",
+              });
+            } else {
+              createToast({
+                title: `Transfer Shipment ${mutationResult.data?.createShipment?.id}`,
+                type: "success",
+                message: "Successfully created a new shipment",
+              });
+
+              navigate(`/bases/${baseId}/transfers/shipments`);
+            }
+          })
+          .catch((err) => {
+            triggerError({
+              message: "Error while trying to create a new shipment!",
+              statusCode: err.code,
+            });
+          });
+      }
+    },
+    [
+      acceptedTransferAgreementsPartnerData,
+      baseId,
+      createShipmentMutation,
+      createToast,
+      triggerError,
+      navigate,
+    ],
+  );
+
+  // Handle Loading State
+  if (allAcceptedTransferAgreements.loading) {
+    return <APILoadingIndicator />;
+  }
+
+  if (
+    partnerOrganisationBaseData?.length === 0 ||
+    partnerOrganisationBaseData === undefined ||
+    allAcceptedTransferAgreements.error
```

Comment:
@aerinsol / @HaGuesto This is the network error you're referring to? in the todo below. just got to double confirm.